File: frontend/src/api/trigger.ts
import api from './index'

export interface TriggerConfig {
  id: string
  user_id: string
  is_enabled: boolean
  inactivity_days: number
  check_in_interval_days: number
  alert_t0_push: boolean
  alert_t3_sms: boolean
  alert_t7_contact: boolean
  last_check_in: string | null
  trigger_started_at: string | null
  trigger_completed_at: string | null
  is_triggered: boolean
  is_emergency_recalled: boolean
  created_at: string
  updated_at: string
}

export interface TriggerStatus {
  is_triggered: boolean
  is_emergency_recalled: boolean
  days_since_last_checkin: number
  inactivity_days: number
  trigger_stage: string
  next_alert_at: string | null
}

export interface TriggerLog {
  id: string
  user_id: string
  event_type: string
  description: string
  metadata_json: string
  created_at: string
}

export const triggerApi = {
  getConfig() {
    return api.get<TriggerConfig>('/trigger/config')
  },
  updateConfig(data: Partial<TriggerConfig>) {
    return api.put<TriggerConfig>('/trigger/config', data)
  },
  checkIn() {
    return api.post<{ message: string; last_check_in: string }>('/trigger/check-in')
  },
  getStatus() {
    return api.get<TriggerStatus>('/trigger/status')
  },
  emergencyRecall() {
    return api.post<{ message: string; is_emergency_recalled: boolean }>('/trigger/emergency-recall')
  },
  getLogs() {
    return api.get<TriggerLog[]>('/trigger/logs')
  },
}

File: backend/app/schemas/trigger.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class TriggerConfigUpdate(BaseModel):
    """更新生命开关配置请求（所有字段可选）"""
    is_enabled: Optional[bool] = None
    inactivity_days: Optional[int] = None
    check_in_interval_days: Optional[int] = None
    alert_t0_push: Optional[bool] = None
    alert_t3_sms: Optional[bool] = None
    alert_t7_contact: Optional[bool] = None

class TriggerConfigResponse(BaseModel):
    """生命开关配置响应"""
    id: str
    user_id: str
    is_enabled: bool
    inactivity_days: int
    check_in_interval_days: int
    alert_t0_push: bool
    alert_t3_sms: bool
    alert_t7_contact: bool
    last_check_in: Optional[datetime] = None
    trigger_started_at: Optional[datetime] = None
    trigger_completed_at: Optional[datetime] = None
    is_triggered: bool
    is_emergency_recalled: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

class CheckInResponse(BaseModel):
    """安全打卡响应"""
    message: str
    last_check_in: datetime

class TriggerLogResponse(BaseModel):
    """触发事件日志响应"""
    id: str
    user_id: str
    event_type: str
    description: str
    metadata_json: str
    created_at: datetime

    model_config = {"from_attributes": True}

class EmergencyRecallResponse(BaseModel):
    """紧急撤回响应"""
    message: str
    is_emergency_recalled: bool

class TriggerStatusResponse(BaseModel):
    """生命开关状态响应"""
    is_triggered: bool
    is_emergency_recalled: bool
    days_since_last_checkin: int
    inactivity_days: int
    trigger_stage: str
    next_alert_at: Optional[str] = None
